test/object_store/test_basic.py

```diff
@@ -53,9 +53,14 @@ def run_with_dir(run_cmd_gen, run_dir):
         log = os.path.join(run_dir, 'log')
         log_fd = os.open(log, os.O_WRONLY | os.O_CREAT | os.O_APPEND, mode=0o666)
         # redirect stdout and stderr to log file
-        for output in [sys.stdout, sys.stderr]:
+        # we have to hardwire the STDOUT_FILENO and STDERR_FILENO. because
+        # pytest mutates sys.stdout and sys.stderr if it captures them, so,
+        # if we intent to redirect the child process's stdout and stderr fd
+        # to the specified fd, we have to use the fd numbers *used* by the
+        # child process, not the ones used by the test.
+        outputs = [(sys.stdout, 1), (sys.stderr, 2)]
+        for output, output_fd in outputs:
             output.flush()
```

Comment:
I wonder if this flush still required

Reply:
it was recommended by Nadav when i migrated this test to pytest. IIUC, he mentioned that without it, the output would be interleaved.

Reply:
see https://github.com/scylladb/scylladb/pull/14165#discussion_r1252641346

Reply:
Flush is needed before fork() and it's still there. Here it flushes whatever sits in child process buffers that are connected to some descriptors, not necessarily 1 and 2. Also -- there's execve() coming up here that would just throw away any memory from current process. I think it was pointless _here_ before this patch and it's still such after it.

Reply:
#15568